spec/behavior/exchangeFsm.spec.js

```diff
@@ -104,19 +104,23 @@ describe( "Exchange FSM", function() {
 
 			var ex = exchangeFn( options );
 			channelMock = ex.mock;
-			var deferred = when.defer();
+
+      const p = new Promise((resolve, reject) => {
+
+      });
+
 			channelMock
 				.expects( "define" )
-				.once()
-				.returns( deferred.promise );
+        .once()
+        .returns(Promise.reject( new Error( "nope" ) ));
```

Comment:
@stheuer : why are you rejecting here? shouldn't it be returning just 'p'?

Reply:
because of original line
```
118 	deferred.reject( new Error( "nope" ) );
```
that makes the  promise reject

Reply:
originally it was
`.returns( deferred.promise );`
which looks like that it was just returning the whole promise. without resolving or rejecting...